src/Middleware/HeaderPropagation/src/HeaderPropagationState.cs

```diff
@@ -0,0 +1,19 @@
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.HeaderPropagation
+{
+    public class HeaderPropagationState
+    {
+        private static AsyncLocal<Dictionary<string, StringValues>> _headers { get; } = new AsyncLocal<Dictionary<string, StringValues>>();
+
+        public Dictionary<string, StringValues> Headers
```

Comment:
@Tratcher - is there another type we want to use for this or is this the best collection type for headers?

Reply:
I was thinking about `IHeaderDictionary` but it's not obvious we want to use that here.

Reply:
Should this property be internal?

Reply:
No, I think this is fine.